Q-9.
Given the score of students in multiple exams as: data Test the hypothesis that the mean scores of all the students are the same. If not, name the student with the highest score.

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [2]:
#Reading the data
data={'Name':['Karan', 'Deepa', 'Karthik', 'Chandan', 'Jeevan'],
                  'Exam_1':[85, 70, 90, 75, 95],
                  'Exam_2':[90, 80, 85, 70, 92],
                  'Final_Exam':[92, 85, 88, 75, 96]}
df = pd.DataFrame(data)
df.head()

Name  Exam_1  Exam_2  Final_Exam
0    Karan      85      90          92
1    Deepa      70      80          85
2  Karthik      90      85          88
3  Chandan      75      70          75
4   Jeevan      95      92          96

Since we must check the mean scores (numeric variable) of all students (categoric variable), we will perform a One-way ANOVA test on the given dataset.

In [3]:

f_value, p_value = stats.f_oneway(df.Exam_1, df.Exam_2, df.Final_Exam)

print("One-way ANOVA:")
print("F-value:", f_value)
print("p-value:", p_value)

One-way ANOVA:
F-value: 0.3236947791164659
p-value: 0.7295954253321559


Since we have found  the p-value (0.7295954253321559) is greater than the common significance level of 0.05, we do not have sufficient evidence to reject the null hypothesis. This suggests that there is no significant difference in the mean scores between the groups (i.e., the mean scores of all the students are considered to be the same).